src/lib/homestore_backend/hs_pg_manager.cpp

```diff
@@ -193,4 +206,73 @@ void HSHomeObject::persist_pg_sb() {
     }
 }
 
+bool HSHomeObject::_get_stats(pg_id_t id, PGStats& stats) const {
+    auto lg = std::shared_lock(_pg_lock);
+    auto it = _pg_map.find(id);
+    if (_pg_map.end() == it) { return false; }
+    auto const& pg = it->second;
+    auto hs_pg = static_cast< HS_PG* >(pg.get());
+    auto const blk_size = hs_pg->repl_dev_->get_blk_size();
+
+    stats.id = hs_pg->pg_info_.id;
+    stats.replica_set_uuid = hs_pg->pg_info_.replica_set_uuid;
+    stats.num_members = hs_pg->pg_info_.members.size();
+    stats.total_shards = hs_pg->total_shards();
+    stats.open_shards = hs_pg->open_shards();
+
+    for (auto const& m : hs_pg->pg_info_.members) {
+        stats.members.emplace_back(m.id, m.name);
+    }
+
```

Comment:
I think we can add a `hs_pg->avail_chunks()` which calls `chunk_selector().avail_chunks(pdev_id)`
and
```
uint32_t HeapChunkSelector:: avail_chunks(pdev_id) {
            return  m_per_dev_heap[pdev_id].second().size();
}
```